# People on Banknotes

Whose faces appear on banknotes?

The file `people-on-banknotes.csv` contains data about individuals featured on banknotes from 38 countries. This dataset spans all 22 subregions and sub-subregions of the world, as defined by the United Nations Statistics Division's geoscheme.

It profiles 241 people, detailing their occupations and the year they first appeared on a banknote. Additionally, it includes their year of death — or `NaN` if they were still alive when the dataset was compiled.

Most banknotes were issued after the featured individual’s death. The column `first_death_diff` calculates the difference between the year of their first appearance on a banknote and their year of death (or remains `NaN` if the person was still living at the time of curation).




**NOTE: This project originated from an open-ended project in Cisco Networking Academy's *Data Science Essentials with Python: Coding for Answers* course. I used the provided project as a starting point and independently extended the analysis.**

In [1]:
import pandas as pd

df = pd.read_csv('people-on-banknotes.csv')
df

,country,currency,name,gender,occupation,value,first_appearance,death,first_death_diff,currency_code
0,Argentina,Argentine Peso,Eva Perón,F,Activist,100,2012,1952,60.0,ARS
1,Argentina,Argentine Peso,Julio Argentino Roca,M,Head of Gov't,100,1988,1914,74.0,ARS
2,Argentina,Argentine Peso,Domingo Faustino Sarmiento,M,Head of Gov't,50,1999,1888,111.0,ARS
3,Argentina,Argentine Peso,Juan Manuel de Rosas,M,Politician,20,1992,1877,115.0,ARS
4,Argentina,Argentine Peso,Manuel Belgrano,M,Founder,10,1970,1820,150.0,ARS
...,...,...,...,...,...,...,...,...,...,...
274,Venezuela,Venezuelan Bolivar,Francisco de Miranda,M,Military,200,1968,1816,152.0,VES
275,Venezuela,Venezuelan Bolivar,Simón Rodrigues,M,Educator,20,2007,1854,153.0,VES
276,Venezuela,Venezuelan Bolivar,Ezequiel Zamora,M,Military,100,2018,1860,158.0,VES
277,Venezuela,Venezuelan Bolivar,Rafael Urdaneta,M,Head of Gov't,10,2018,1845,173.0,VES


### Quick cleaning

The same person can appear on multiple banknotes. Below we drop the `value` column and remove duplicate people.

In [2]:
df = df.drop(columns=['value'])
df = df.drop_duplicates(subset="name")
df

,country,currency,name,gender,occupation,first_appearance,death,first_death_diff,currency_code
0,Argentina,Argentine Peso,Eva Perón,F,Activist,2012,1952,60.0,ARS
1,Argentina,Argentine Peso,Julio Argentino Roca,M,Head of Gov't,1988,1914,74.0,ARS
2,Argentina,Argentine Peso,Domingo Faustino Sarmiento,M,Head of Gov't,1999,1888,111.0,ARS
3,Argentina,Argentine Peso,Juan Manuel de Rosas,M,Politician,1992,1877,115.0,ARS
4,Argentina,Argentine Peso,Manuel Belgrano,M,Founder,1970,1820,150.0,ARS
...,...,...,...,...,...,...,...,...,...
274,Venezuela,Venezuelan Bolivar,Francisco de Miranda,M,Military,1968,1816,152.0,VES
275,Venezuela,Venezuelan Bolivar,Simón Rodrigues,M,Educator,2007,1854,153.0,VES
276,Venezuela,Venezuelan Bolivar,Ezequiel Zamora,M,Military,2018,1860,158.0,VES
277,Venezuela,Venezuelan Bolivar,Rafael Urdaneta,M,Head of Gov't,2018,1845,173.0,VES


### Project Ideas

- What proportion of individuals featured are male versus female?

- Are writers or politicians more commonly depicted?

- What percentage of featured individuals are musicians?

- What percentage of banknotes were issued before the person’s death?

- Who is the oldest historical figure in the dataset?

- Which countries feature the oldest historical figures on their banknotes?

- What percentage of individuals died at least 100 years before appearing on a banknote?

- Which individuals appeared on a banknote just one year after their death?


**QUESTION 1: What proportion of individuals featured are male versus female?**

In [3]:
gen_prop = df['gender'].value_counts(normalize=True) * 100 #returns the proportion instead of raw count
gen_prop

gender
M    78.008299
F    21.991701
Name: proportion, dtype: float64

**ANSWER:** Approximately 78% of the featured persons are male, and around 21.99% are female.

**QUESTION 2: Are writers or politicians more commonly depicted?**

In [4]:
occu_prop = (df['occupation'].value_counts(normalize=True) * 100).sort_values(ascending= False)
occu_prop

occupation
Writer              18.672199
Head of Gov't       15.767635
Politician          11.203320
Founder             10.373444
Revolutionary        8.713693
STEM                 6.639004
Military             5.394191
Visual Artist        5.394191
Musician             4.979253
Monarch              4.979253
Other                2.074689
Activist             1.659751
Educator             1.659751
Religious figure     1.244813
Performer            1.244813
Name: proportion, dtype: float64

**ANSWER:** Yes, writers are the most commonly depicted, while politicaians ranks third.

**QUESTION 3: What percentage of featured individuals are musicians?**

**ANSWER:** Approximately 4.98% of the featured individuals are musicians.

**QUESTION 4: What percentage of banknotes were issued before the person’s death?**

In [39]:
bef_death = (df['first_death_diff'] < 0).mean() * 100 
bef_death

np.float64(3.7344398340248963)

**ANSWER:** Approximately 3.73% of the banknotes were issued before the individual's death.

**QUESTION 5: Who is the oldest historical figure in the dataset?**

In [29]:
oldest = df.sort_values('death', ascending=True).head(1)
oldest

,country,currency,name,gender,occupation,first_appearance,death,first_death_diff,currency_code,bef_death
234,Tunisia,Tunisian Dinar,Hannibal,M,Military,2013,-183,2196.0,TND,False


**ANSWER:** The oldest historical figure that appeared on a banknote is Hannibal in a Tunisian Dinar currency. He died in 183 BC and first appeared in a banknote in 2013.

**QUESTION 6: Which countries feature the oldest historical figures on their banknotes?
	- Hint: Group by country and aggregate the year of death using the median. Sort the results.**

In [34]:
#convert first the death column from str to int
df['death'] = df['death'].fillna(0).astype(int)

#proceed coding for the median death year by country
country_oldest = df.groupby('country')['death'].median().reset_index().sort_values('death', ascending=True).head(10)
country_oldest

,country,death
29,South Korea,1560.5
31,São Tomé and Príncipe,1779.5
34,Ukraine,1817.5
35,United States,1826.0
3,Bolivia,1830.0
14,Iceland,1845.0
37,Venezuela,1849.5
10,Czech Republic,1869.0
6,Chile,1879.0
0,Argentina,1888.0


**ANSWER:** The top 5 countries that featured the oldest historical figures are South Korea, São Tomé and Príncipe, Ukraine, United States, and Bolivia.

**QUESTION 7: What percentage of individuals died at least 100 years before appearing on a banknote?**

In [37]:
more_hund = (df['first_death_diff'] >= 100).mean() * 100
more_hund

np.float64(33.19502074688796)

**ANSWER:** About 33.20% of the individuals died at least 100 years before they appreared on a banknote. 

**QUEESTION 8: Which individuals appeared on a banknote just one year after their death?**

In [43]:
one_year_death = df.query('first_death_diff == 1')
one_year_death

,country,currency,name,gender,occupation,first_appearance,death,first_death_diff,currency_code,bef_death
63,Colombia,Colombian Peso,Gabriel García Márquez,M,Writer,2015,2014,1.0,COP,False
173,Nigeria,Nigerian Naira,General Murtala Mohammed,M,Head of Gov't,1977,1976,1.0,NGN,False
190,Philippines,Philippine Piso,Corazon C. Aquino,F,Head of Gov't,2010,2009,1.0,PHP,False
191,Philippines,Philippine Piso,Manuel A. Roxas,M,Head of Gov't,1949,1948,1.0,PHP,False


**ANSWER:** Grabriel Garcia Marquez, General Murtula Mohammed, Corazon C. Aquino, and Manuel A. Roxas died just 1 year after they appeared in a banknote.